749227 - net-snmp dies with "select: bad file descriptor" when Agentx connection is closed

commit 434bfc1f7457550b6f87d5bcd511276b270b5964
Author: Jan Safranek <jsafranek@users.sourceforge.net>
Date:   Mon Nov 14 11:22:53 2011 +0100

    CHANGES: snmpd: from Martin Wilck: fixed snmpd crashing with 'select: bad file descriptor'.
    
    On 32-bit systems, snmpd dies with aforementioned message when using file descriptions > 32, e.g. when having >32 AgentX subagents.

diff --git a/include/net-snmp/library/large_fd_set.h b/include/net-snmp/library/large_fd_set.h
index 21f473c..e24c34f 100644
--- a/include/net-snmp/library/large_fd_set.h
+++ b/include/net-snmp/library/large_fd_set.h
@@ -71,7 +71,7 @@ int    netsnmp_large_fd_is_set(SOCKET fd, netsnmp_large_fd_set *fdset);
  * According to SUSv2, this array must have the name fds_bits. See also
  * <a href="http://www.opengroup.org/onlinepubs/007908775/xsh/systime.h.html">The Single UNIX Specification, Version 2, &lt;sys/time.h&gt;</a>.
  */
-#define NETSNMP_FD_MASK_SIZE sizeof(((fd_set*)0)->fds_bits)
+#define NETSNMP_FD_MASK_SIZE sizeof(((fd_set*)0)->fds_bits[0])
 
 /** Number of bits in one element of the fd_set::fds_bits array. */
 #define NETSNMP_BITS_PER_FD_MASK (8 * NETSNMP_FD_MASK_SIZE)
